### PR TITLE
Replace E2E Server docker instance with live

### LINKED ISSUE #3293: Replace WCPay Local Server Instance on E2E with WCPay Live Server
At the moment, for WCPay E2E Test Suite, we're running a local instance of the WCPay Server, which leads to problems such as latency in response from Stripe, webhooks reliability and security concerns discussed here - paJDYF-2PM-p2#comment-8617

Based on the discussion, if we move E2E tests to run on schedule instead of running it on each PR, it would be better to replace the local server instance with live.

### PR DESCRIPTION
**Pull request by achyuthajoy**
Fixes #3293 

### Changes proposed in this Pull Request

Based on the discussion on this thread paJDYF-2PM-p2#comment-8617, running WCPay Server instance in GH Actions has security & latency issues. 

Changes:
* Skips creation of WCPay server docker instance for running E2E Tests and uses Live server on GH Actions. 
* Allows running WCPay server docker instance locally along with the option to use Live server via declaring local env variables.
* E2E Tests Documentation update

### Testing instructions

**Testing GH Actions**

* Go to https://github.com/Automattic/woocommerce-payments/actions/workflows/e2e-test.yml
* Click `Run workflow` and choose the branch `e2e-optim/use-live-server` to trigger the E2E tests manually
* Choose the created test suite -> Choose test -> Look for the step `Run npm run test:e2e`
* Confirm that the server instance is not created. [Old log](https://user-images.githubusercontent.com/1161872/141801626-8481a1c9-dd28-4959-98b4-0024a2790670.png) | [New log](https://user-images.githubusercontent.com/1161872/141801918-6258b342-bb7a-4427-910a-1b38258079ae.png)

**Testing Local**

* Following the instructions in this [README.pdf](https://github.com/Automattic/woocommerce-payments/files/7603465/README.pdf) (Copy of the documentation) and try running local server & live server. 
* Confirm that the changes works as expected.

-------------------

- [ ] Added changelog entry (or does not apply)
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

* Remove repo secrets related to WCPay Server setup (@achyuthajoy)

- [ ] Added testing instructions to the [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) (or does not apply)

### CODE REVIEW

**Comment by deepakpathania** [comment]
Do we need to update any associated documentation for the E2E tests to reflect this change?

**Comment by deepakpathania** [comment]
Can we add a benchmark for the action with and without these changes for context in the description?

**Comment by deepakpathania** [comment]
Verified that the setup script no longer has the server setup steps

Screenshot 2021-11-16 at 11 52 04 AM
.

**Comment by ricardo** [comment]
@achyuthajoy, @deepakpathania, Does this need someone else to review?

I'm janitor this week taking a look at E2E PRs. Please LMK if this should be reviewed by me 👍

**Comment by deepakpathania** [comment]
> Does this need someone else to review?

Yes, it'll be helpful to get another pair of eyes on this. Thanks @ricardo

**Comment by achyuthajoy** [comment]
Hey @ricardo I'm working on a few changes on this PR. Will let you know once it's ready for review.

**Comment by achyuthajoy** [comment]
@deepakpathania @ricardo I've made a few changes to the PR and updated the testing instructions. You can go ahead with the review :)

File: tests/e2e/README.md
Comment (deepakpathania) [inline tests/e2e/README.md:34]: Could we use this as true by default so that folks don't have to change their local config after pulling in these changes?

Comment (achyuthajoy) [reply tests/e2e/README.md:34]: Sure @deepakpathania I'll update the docs

File: tests/e2e/README.md
Comment (deepakpathania) [inline tests/e2e/README.md:54]: We'll have to set `E2E_USE_LOCAL_SERVER` as `false` for this to work, right?

Comment (achyuthajoy) [reply tests/e2e/README.md:54]: @deepakpathania I'm making changes to make local server default. So once it's ready, you need to explicitly set `E2E_USE_LOCAL_SERVER` to `false` to use the live server

Comment (deepakpathania) [reply tests/e2e/README.md:54]: Yes, so we should mention that here in the required ENV variables list for using the live server. I just switched from local instance to live server locally and had to explicitly set it up for the clean up to work as intended.

**Comment by deepakpathania** [comment]
- Confirmed that the setup script no longer has the server setup steps.
- Verified that setting `E2E_USE_LOCAL_SERVER=true` uses the local server instance.
- Verified that the live server is used with the correct env variables.

Left a comment above how `E2E_USE_LOCAL_SERVER=false` needs to be set for using the live server instance. @achyuthajoy

File: tests/e2e/README.md
Comment (deepakpathania) [inline tests/e2e/README.md:46]: Nitpick, `you'll need to *add*  Jetpack blog token*`

Comment (achyuthajoy) [reply tests/e2e/README.md:46]: 😄 Nice find

File: tests/e2e/README.md
Comment (achyuthajoy) [reply tests/e2e/README.md:60]: Made the changes :)

**Comment by achyuthajoy** [comment]
@deepakpathania I've made the suggested changes and updated the docs. Uploaded the new docs PDF as well

**Comment by deepakpathania** [comment]
LGTM. Thanks for your hard work. 🚢 

We could wait for @ricardo to have a look once before merging, approving from my end for now.

**Review by deepakpathania — APPROVED** [review]

File: tests/e2e/env/setup.sh
Comment (ricardo) [inline tests/e2e/env/setup.sh:42]: This is not related to the issue at hand, so feel free to ignore, but it's nice to fix it sooner or later:

- The test and live `CLIENT_ID`s were removed from the server, they're no longer needed.
- The `WCPAY_OAUTH_ENCRYPT_KEY` got renamed to `WCPAY_ONBOARDING_ENCRYPT_KEY`.

**Review by ricardo — APPROVED** [review]
LGTM.

Just added a minor comment about some outdated constants.

File: tests/e2e/env/setup.sh
Comment (deepakpathania) [reply tests/e2e/env/setup.sh]: Addressed in [8111110](https://github.com/Automattic/woocommerce-payments/pull/3383/commits/81111101f2f48c954f471829ba01ad5c38a29168)

**Pull request merged**
